src/lib.rs

```diff
@@ -10,28 +10,95 @@
 // distributed under the License is distributed on an "AS IS" BASIS,
 // See the License for the specific language governing permissions and
 // limitations under the License.
+#![recursion_limit = "128"]
+#![type_length_limit = "2097152"]
```

Comment:
Looks like actually building the crate instead of `cargo check` or `cargo clippy` would actually trigger the error.
```bash
error: reached the type-length limit while instantiating `<std::vec::IntoIter<futures::future::and_then::AndThen<futures::...`                                                                                     
  |                                                                                                                                                                                                                
  = note: consider adding a `#![type_length_limit="2097152"]` attribute to your crate                                                                                                                              
                                                                                                                                                                                                                   
error: aborting due to previous error                                                                                                                                                                              
                                                                                                                                                                                                                   
error: Could not compile `tikv-client`.                                                                                                                                                                            
```
After trying different numbers I managed to change it to a smaller number.
```rust
#![type_length_limit = "1572864"]
```

Reply:
Nice digging. :)